simplivity/connection.py

```diff
@@ -79,8 +79,7 @@ def do_http(self, method, path, body, custom_headers=None, login=False):
             http_headers.update(custom_headers)
 
         try:
-            if not self.__connection:
-                self.__connection = self.get_connection()
+            self.__connection = self.get_connection()
```

Comment:
Working on it

Reply:
Can you clarify the package was caching the connection to start with?

I was reading the pydocs and I found this.

An HTTPConnection instance represents one transaction with an HTTP server. It should be instantiated passing it a host and optional port number. If no port number is passed, the port is extracted from the host string if it has the form host:port, else the default HTTP port (80) is used. 

In order testing we are hitting two primary issues:

*1 If the HTTPSConnection Object is inactive for more than 120 seconds we experience a RemoteDisconnection Exception
*2 If Threading we occasionally experience a CannotSendRequest Exception because the connection is being cache for every do_http request instead of a new connection for each unique request.

Reply:
Caching is implemented to avoid making a connection for every API calls. Yes, as per the doc the connection can be used for a single transaction but I have tested this with multiple successful calls (maybe in less than 120 sec).
Agree, in case of threading it can be a problem.
Let's go ahead without caching now, let me think if any other ways to reuse the connection.